assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -0,0 +1,617 @@
+        // This case won't delete them.
+        delete($scope.updatedBuildConfig.spec.output.to)
+      } else {
+        $scope.updatedBuildConfig.spec.output.to = to;
+      }
+
+      // Update envVars
+      var updateEnvVars = [];
+      angular.forEach($scope.envVars, function(v, k) {
+        var env = {
+          name: k,
+          value: v
+        };
+        updateEnvVars.push(env);
+      });
+      $filter('buildStrategy')($scope.updatedBuildConfig).env = updateEnvVars
+
+      // Update triggers
+      var triggers = [];
+      if ($scope.triggers.webhook) {
+        var webhooks = _.filter($scope.buildConfig.spec.triggers, function(obj) { return obj.type == "GitHub" })
+        if (webhooks.length === 0) {
+          webhooks.push({
+            github: {
+              secret: ApplicationGenerator._generateSecret()
+            },
+            type: "GitHub"
+          });
+        }
+        triggers = triggers.concat(webhooks);
+      }
+      triggers = triggers.concat(_.filter($scope.buildConfig.spec.triggers, function(obj) { return obj.type == "Generic" }));
+
+      if ($scope.triggers.imageChange) {
+        var imageChangeTriggers = _.filter($scope.buildConfig.spec.triggers, function(obj) { return obj.type == "ImageChange" });
+        if (imageChangeTriggers.length === 0) {
+          imageChangeTriggers.push({
+            imageChange: {},
+            type: "ImageChange"
+          });
+        }
+        triggers = triggers.concat(imageChangeTriggers);
+      }
+      if ($scope.triggers.configChange) {
+        triggers.push({
+          type: "ConfigChange"
+        });
+      }
+      $scope.updatedBuildConfig.spec.triggers = triggers;
+
+      DataService.update("buildconfigs", $scope.updatedBuildConfig.metadata.name, $scope.updatedBuildConfig, {
+        namespace: $scope.updatedBuildConfig.metadata.namespace
+      }).then(
+        function() {
+          AlertMessageService.addAlert({
+            name: $scope.updatedBuildConfig.metadata.name,
+            data: {
+              type: "success",
+              message: "Build Config " + $scope.updatedBuildConfig.metadata.name + " was sucessfully updated."
+            }
+          });
+          $location.path(Navigate.resourceURL($scope.updatedBuildConfig, "BuildConfig", $scope.updatedBuildConfig.metadata.namespace, "browse"));
+        },
+        function(result) {
+          $scope.disableInputs = false;
+          AlertMessageService.addAlert({
+            name: $scope.updatedBuildConfig.metadata.name,
+            data: {
+              type: "error",
+              message: "An error occurred updating the build " + $scope.updatedBuildConfig.metadata.name + "Build Config",
+              details: $filter('getErrorDetails')(result)
+            }
+          });
+          $location.path(Navigate.resourceURL($scope.updatedBuildConfig, "BuildConfig", $scope.updatedBuildConfig.metadata.namespace, "browse"));
```

Comment:
I think it would be better to stay on the page to give the user a chance to fix the problem (assuming something is wrong with the data) rather than navigating away and losing any edits.